tests/e2e/volumecreate_test.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test is going to cover various volume creation test cases
+
+package e2e
+
+import (
+	"log"
+	"os"
+	"strings"
+
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/govc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+
+	. "gopkg.in/check.v1"
+)
+
+const ErrorVolumeCreate = "Error response from daemon: create"
+
+type VolumeCreateTestSuite struct {
+	hostIP     string
+	esxIP      string
+	dsNameList []string
+	volumeList []string
+}
+
+func (s *VolumeCreateTestSuite) SetUpSuite(c *C) {
+	s.hostIP = os.Getenv("VM2")
+	s.esxIP = os.Getenv("ESX")
+	s.dsNameList = govc.GetDatastoreList()
+	log.Printf("Datastores found are %v", s.dsNameList)
+}
+
+func (s *VolumeCreateTestSuite) SetUpTest(c *C) {
+	s.volumeList = s.volumeList[:0]
+}
+
+func (s *VolumeCreateTestSuite) TearDownTest(c *C) {
+	for _, name := range s.volumeList {
```

Comment:
I would suggest to create a volume name string with `space` separated and pass down to deleteVolume() function .. something like `docker volume rm vol1 vo2 ` that avoids subsequent ssh calls... get the response of the command and look for any `error` string . if not present we are good to go.